src/d3a/models/strategy/external_strategy.py

```diff
@@ -0,0 +1,150 @@
+"""
+Copyright 2018 Grid Singularity
+This file is part of D3A.
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""
+
+from d3a.models.strategy import BaseStrategy
+
+import json
+from d3a.models.market import MarketRedisApi
+
+
+class RedisExternalConnection(MarketRedisApi):
+    def __init__(self, area):
+        self.area = area
+        super().__init__(None)
+        self.areas_to_register = []
+
+    @property
+    def market(self):
+        return self.area.next_market
+
+    @property
+    def _offer_channel(self):
+        return f"{self.area.parent.slug}/{self.area.slug}/offer"
+
+    @property
+    def _delete_offer_channel(self):
+        return f"{self.area.parent.slug}/{self.area.slug}/delete_offer"
+
+    @property
+    def _accept_offer_channel(self):
+        return f"{self.area.parent.slug}/{self.area.slug}/accept_offer"
```

Comment:
May be offline if you can help me understand the reason to overwrite these method of base class __MarketRedisApi__

Reply:
Simple, the external connection needs to expose different channels than the simulation itself (eg. from an Area to its market). We can discuss this offline too if you want